### ISSUE TITLE
please hint about missing "as()" when from(subquery) is missing it

### ISSUE THREAD

**Issue by kapouer** [knex/knex#3485]
# Environment

Knex version: 0.19.5
Database + version: postgresql 11
OS: linux/debian

# Feature discussion / request

I find myself often spending hours trying to figure how to make
```
knex('table').select(rawstr).from(knex('table').where(obj))
```
work.
It's very simple:
```
knex('table').select(rawstr).from(knex('table').where(obj).as('sub'))
```

However knex outputs a wrong sql statement (missing parenthesis) and when debugging, it looks like a bug in knex.

I think it would be nice to users if a missing as() could be detected and a warning printed.

**Comment by edvardchen** [comment]
Anyone work no it ? If not, I can take it

**Comment by kapouer** [comment]
@edvardchen that'd be nice !

**Comment by edvardchen** [comment]
@kapouer kinda busy this week. I would do it in the weekend. Sorry to reply so late

**Comment by edvardchen** [comment]
Almost done. I would add more test before submitting PR

**Comment by edvardchen** [comment]
All dialects except 'sqlite3' require that subquery must have an alias.

1. Should we add an anonymous alias for them like `T`,
2. or just let the dialects to complain?
    The error messages thrown from dialects are really comprehensible. 

@kapouer @elhigu @kibertoad

**Comment by kibertoad** [comment]
@edvardchen What's the advantage of anonymous aliases, if they are not referenced anywhere in the query anyway? If messages are explicit enough for users to understand what is happening, I think that's good enough.

**Comment by edvardchen** [comment]
Anonymous aliases can make the query executable even not being referenced. Like:
```sql
select id from (select id from my_tab) as T
```
But I worry that it may not be expected and would cause confusions

**Comment by kapouer** [comment]
[An interesting thread to read](https://www.postgresql.org/message-id/flat/CAKFQuwYYY4OyJ2Z39v9NyZGwcEAGHRvymJ24zJhAjiV1u4F4pA%40mail.gmail.com#06869c49dceb228c92be857b12c6f0a4) about bringing anonymous aliases to postgresql.
@edvardchen the main problem seems to be sure anonymous aliases won't conflict with anything.
Maybe it's safer stick to whatever SQL dialect is available.

**Comment by edvardchen** [comment]
OK.

**Comment by kibertoad** [comment]
Released in 0.20.1
